### PR TITLE
chore: hint to set requirement to regex

### PR DESCRIPTION
**Pull request by koozz**
### Why:

Ran into an issue about the new rulesets and enforcing Conventional Commits. If the requirement field is not set to match a regex pattern but rather to 'start with a matching pattern', it will not work.

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Added a hint about setting the right requirement when working with regular expressions.

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by koozz** [comment]
Preview looks good: https://docs-25279-d9643f.preview.ghdocs.com/en/enterprise-cloud@latest/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/managing-rulesets-for-a-repository#useful-regular-expression-patterns

**Comment by cmwilson21** [comment]
@koozz Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: data/reusables/repositories/rulesets-commit-regex.md [hunk 1]
```diff
@@ -16,6 +16,8 @@ The negative lookahead assertion, denoted `?!`, is not supported. However, for c
 
 The following examples provide useful patterns for commit metadata.
 
+Make sure to set the requirement field to: `Must match a given regex pattern`.
```
Comment (isaacmbrown) [inline data/reusables/repositories/rulesets-commit-regex.md:19]:
```suggestion
The following examples provide useful patterns for commit metadata. To use these patterns, set **Requirement** to "Must match a given regex pattern".
```

**Review by isaacmbrown — APPROVED** [review]
Hey @koozz, thanks for the suggestion, this is a good callout to add! I just applied a couple of styling changes to match our usual tone, and I'll get this merged now 🚀

**Comment by tarat69** [comment]
Keep hacking me

**Pull request merged**